File: Frequency Game[31-May-2023]/Solution.py
#  >> PYTHON CODE <<

def LargButMinFreq(arr,n):
    mp={}
    
    for element in arr:
        mp[element]=mp.get(element,0)+1

    ans=-1e9
    freq=1e9
    
    #print(mp)
    for key in mp:
        if(mp.get(key)<freq):
            ans=key
            freq=mp.get(key)
        elif mp.get(key)==freq:
            ans=max(key,ans)
    
    return ans

File: Frequency Game[31-May-2023]/Solution.java
//  >> JAVA CODE <<

public static int LargButMinFreq(int arr[], int n) {
        var map=new HashMap<Integer,Integer>();
        
        for(var el:arr)
        {
            map.put(el,map.getOrDefault(el,0)+1);
        }
        
        int ans=Integer.MIN_VALUE;
        int freq=Integer.MAX_VALUE;
        for(var key:map.keySet())
        {
            if(map.get(key)<freq)
            {
                ans=key;
                freq=map.get(key);
            }
            else if(map.get(key)==freq)
            {
                ans=Math.max(ans,key);
            }
        }
        
            return ans;
    }